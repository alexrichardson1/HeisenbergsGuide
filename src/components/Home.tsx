import "styles/Home.css";
import { LinkContainer } from "react-router-bootstrap";
import { useEffect } from "react";

function App(): JSX.Element {
  useEffect(() => {
    document.title = "Cook Off Guide";
  }, []);

  return (
    <div>
      <h1>General Tips &amp; Information</h1>
      <p className="info">
        Grab a full set of ingredients from the locations that will be the most
        difficult to return to. The locations by most difficult to get to and
        return from are as follows:
        <ol>
          <li> White pickup truck </li>
          <li> Shed behind the home / Shed to the left of the home </li>
          <li> Basement</li>
          <li> Bathroom</li>
        </ol>
      </p>
      <p className="info">
        Move Bags when the Van Arrives. It is up to you to determine whether or
        not it is safe to transport the bags.
      </p>
      <p className="info">
        There are multiple preset spots for cloakers to hide:
        <ul>
          <li>
            Behind the <u>Blue Dumpster</u> next to the White Sedan
          </li>
          <li>
            Under the <u>White Sedan</u>
          </li>
          <li>
            In the first or second story of the home (depending on the lab
            location)
          </li>
          <li>Behind the fence near where the van would arrive on Rats</li>
        </ul>
        The Cloakers can hide behind the wall in front of the window in the main
        section of the home (across from the bathroom next to the back door).
        They can also hide behind the countertop.
      </p>
      <p className="info">
        If you have a first-level lab, the Bathroom is an extremely safe spot as
        the Officers are <u>NOT</u> programmed to go into there, and only enter
        on extremely rare occasions. The only real way they go into the bathroom
        is if you lead them in there. If you store bags in the bathroom tub,
        cops will <u>NOT</u> take them, or teleport them.
      </p>
      <p className="info">
        <u>Subtitles</u> are especially helpful, as Bain is often ambiguous or
        uncertain.
      </p>
      <p className="info">
        Having a dedicated cook that knows the lines is of most importance. It
        is better to take a slightly longer amount of time with just one
        dedicated cook, than having someone blow up the lab.
      </p>
      <p className="info">
        If multiple people will cook, then have <u>only ONE</u> person cook per
        batch. Confusion is minimized, and you will know what the last
        ingredient is before Bain says what to add, thereby speeding up the
        cook. Only players who know the lines should be cooks.
      </p>
      <p className="info">
        There are 5 sets of planks in this Heist. If you have an upstairs lab,
        barricade the back room with two and use the remaining three as you
        wish. With a downstairs lab, my recommendation is to place one on the
        window next to the stove. The remaining can be used how you wish.
      </p>
      <p className="info">
        Refrain from using grenades. In nearly any circumstance, refrain from
        using grenades. Not only do you destroy nearly all of the barricades,
        but you also do severe damage to any teammates in the blast radius.
      </p>
      <p className="info">
        With an upstairs lab, focus on fighting within the stairs/bathroom,
        retreating to the back room if needed. If the lab is downstairs, fight
        around the bathroom.
      </p>
      <p className="info">
        <u>Ignore the power</u>. Get used to using a flashlight, or get used to
        it being dark. Whether or not you break the light bulbs is up to you.
      </p>
      <p className="info">
        The helicopter can do one of two things:
        <ol>
          <li> It can deploy a bulldozer</li>
          <li> Spawn yellow SWAT on the balcony</li>
        </ol>
        If hovering behind the home, a bulldozer will deploy from the
        helicopter. If hovering above the home, it will spawn yellow SWAT on the
        balcony.
      </p>

      <LinkContainer to="/dialogue">
        <section className="container">
          <button data-hover="View Dialogue">
            <div>Cook Off Bain&apos;s Dialogues</div>
          </button>
        </section>
      </LinkContainer>
    </div>
  );
}

export default App;
