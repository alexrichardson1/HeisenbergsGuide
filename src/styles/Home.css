h1 {
  font-size: 2em;
  padding: 10px 10px;
  border: 10px solid white;
  display: inline-flex;
  background-color: blue;
  margin-left: 50px;
  margin-right: 50px;
}

.info {
  border: 5px solid white;
  display: inline-block;
  clear: left;
  background-color: blue;
  font-size: 0.75em;
  font-weight: 700;
  text-transform: uppercase;
  letter-spacing: 2px;
  padding: 20px 20px;
  margin-left: 50px;
  margin-right: 50px;
}

*:before,
*:after {
  box-sizing: border-box;
  padding: 0;
  margin: 0;
  font: 300 1em/1.5 "Open Sans", "Helvetica Neue", Arial, sans-serif;
  text-decoration: none;
  color: white;
}

section.container {
  min-width: auto;
  margin: 10% auto;
  text-align: center;
}

button:hover {
  cursor: pointer;
}

button {
  background: transparent;
  outline: none;
  position: relative;
  border: 2px solid white;
  padding: 15px 50px;
  overflow: hidden;
}

button:hover:before {
  opacity: 1;
  transform: translate(0, 0);
}

button:before {
  content: attr(data-hover);
  position: absolute;
  top: 1.1em;
  left: 0;
  width: 100%;
  text-transform: uppercase;
  letter-spacing: 3px;
  font-weight: 800;
  font-size: 0.8em;
  opacity: 0;
  transform: translate(-100%, 0);
  transition: all 0.3s ease-in-out;
}

button:hover div {
  opacity: 0;
  transform: translate(100%, 0);
}

button div {
  text-transform: uppercase;
  letter-spacing: 3px;
  font-weight: 800;
  font-size: 0.8em;
  transition: all 0.3s ease-in-out;
  color: white;
}

@media (min-width: 481px) {
  .info {
    font-size: 0.75em;
    margin-left: 50px;
    margin-right: 50px;
  }

  body {
    background-image: url(../images/masks.jpg);
    background-size: cover;
    background-attachment: fixed;
  }
}

/* handled devices */
@media (max-width: 480px) {
  .info {
    font-size: 1em;
  }

  body {
    background-color: black;
  }
}
