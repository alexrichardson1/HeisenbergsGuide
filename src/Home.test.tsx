import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import Home from "./components/Home";

test("renders correctly", () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <Home />
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
