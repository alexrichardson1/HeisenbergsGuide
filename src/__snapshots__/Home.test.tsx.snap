// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <h1>
    General Tips & Information
  </h1>
  <p
    className="info"
  >
    Grab a full set of ingredients from the locations that will be the most difficult to return to. The locations by most difficult to get to and return from are as follows.
    <ol>
      <li>
         White pickup truck 
      </li>
      <li>
        Shed behind the home / Shed to the left of the home 
      </li>
      <li>
        Basement
      </li>
      <li>
        Bathroom
      </li>
    </ol>
  </p>
  <p
    className="info"
  >
    Move Bags when the Van Arrives; It is up to you to determine whether or not it is safe to transport the bags. If so, move quickly and take your finger off the trigger for the time being, Unless you take enemy fire. (It will slightly increase the time it takes for Law Enforcers to engage you, giving you a better chance of survival. Keep in mind Specials such as Bulldozers, Tasers, Cloakers and Shields are exempt from this tip. Kill them on sight.)
  </p>
  <p
    className="info"
  >
    There are multiple preset spots for cloakers to hide, They will hide behind the Blue Dumpster next to the White Sedan; Under the White Sedan; In the first or second story of the home.(Depending on the lab location.) and behind the fence near where the van would arrive on Rats. (The furthest down the street that the van can arrive.)
  </p>
  <p
    className="info"
  >
    The Cloakers can hide behind the wall in front of the window in the main section of the home (Across from the bathroom next to the back door). They can also hide behind the counter top.
  </p>
  <p
    className="info"
  >
    If you have a first level lab, The Bathroom is an extremely safe spot as the Officers are not programmed to go into there, and In such only enter on extremely rare occasions; The only real way they go into the bathroom is if you lead them in there. If you store bags in the bathroom tub, Cops will 
    <b>
      NOT
    </b>
     take them, or teleport them.
  </p>
  <p
    className="info"
  >
    Subtitles are especially helpful, as Bain is often ambiguous or uncertain.
  </p>
  <p
    className="info"
  >
    Having a dedicated cook that knows the lines is paramount. If it is just one person in the lobby, then have just one dedicated cook. It is better to take a slightly longer amount of time with just 1 cook, than have someone blow the lab. (If you see me in game, I am the designated cook. Only exception is if I am in custody.)
     
  </p>
  <p
    className="info"
  >
    If multiple people will cook, then have just 1 person cook per batch. Confusion is minimized, and you will know what the last ingredient is before Bain says what to add, thereby speeding up the cook. (Once again, Only players who know the lines are to be cooks.)
  </p>
  <p
    className="info"
  >
    There are 5 sets of planks in this Heist; If you have an upstairs lab, Barricade the back room with two and use the remaining three as you wish. With a downstairs lab, My recommendation is to place one on the window next to the stove. The remaining can be used how you wish.
  </p>
  <p
    className="info"
  >
    Refrain from using Grenades. In nearly any circumstance, refrain from using grenades. Not only do you destroy nearly all of the homes barricades, you also do severe damage to any teammates in the blast radius.
  </p>
  <p
    className="info"
  >
    With an upstairs lab, focus on fighting within the stairs/bathroom, retreating to the back room if needed. If the lab is downstairs, fight around the Bathroom.
  </p>
  <p
    className="info"
  >
    Ignore the power. Get used to using a Flashlight, or get used to it being dark. Whether or not you break the light bulbs is up to you.
  </p>
  <p
    className="info"
  >
    The Helicopter can do 1 of 2 things; It can deploy a Bulldozer, or spawn Yellow SWAT on the balcony. If hovering behind the home, A Bulldozer will deploy from the Helicopter. If hovering above the home, it will spawn Yellow SWAT on the balcony.
  </p>
  <section
    className="container"
    href="/dialogue"
    onClick={[Function]}
  >
    <button
      data-hover="View Dialogue"
    >
      <div>
        Cook Off Bain's Dialogues
      </div>
    </button>
  </section>
</div>
`;
